server/src/main/java/io/orkes/conductor/server/service/OrkesWorkflowSweepWorker.java

```diff
@@ -0,0 +1,249 @@
+import com.netflix.conductor.core.exception.NotFoundException;
+import com.netflix.conductor.core.execution.WorkflowExecutor;
+import com.netflix.conductor.core.execution.tasks.SystemTaskRegistry;
+import com.netflix.conductor.core.execution.tasks.WorkflowSystemTask;
+import com.netflix.conductor.core.utils.QueueUtils;
+import com.netflix.conductor.core.utils.Utils;
+import com.netflix.conductor.dao.ExecutionDAO;
+import com.netflix.conductor.dao.QueueDAO;
+import com.netflix.conductor.metrics.Monitors;
+import com.netflix.conductor.model.TaskModel;
+import com.netflix.conductor.model.WorkflowModel;
+import com.netflix.conductor.service.ExecutionLockService;
+
+import io.orkes.conductor.metrics.MetricsCollector;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static com.netflix.conductor.core.config.SchedulerConfiguration.SWEEPER_EXECUTOR_NAME;
+import static com.netflix.conductor.core.utils.Utils.DECIDER_QUEUE;
+
+@Component
+@ConditionalOnProperty(name = "conductor.orkes.sweeper.enabled", havingValue = "true")
+@Slf4j
+public class OrkesWorkflowSweepWorker {
+
+    private final QueueDAO queueDAO;
+    private final ConductorProperties properties;
+    private final WorkflowExecutor workflowExecutor;
+    private final ExecutionDAO executionDAO;
+    private final MetricsCollector metricsCollector;
+    private final SystemTaskRegistry systemTaskRegistry;
+    private final ExecutionLockService executionLockService;
+
+    public OrkesWorkflowSweepWorker(
+            QueueDAO queueDAO,
+            WorkflowExecutor workflowExecutor,
+            ExecutionDAO executionDAO,
+            MetricsCollector metricsCollector,
+            SystemTaskRegistry systemTaskRegistry,
+            ExecutionLockService executionLockService,
+            ConductorProperties properties) {
+        this.queueDAO = queueDAO;
+        this.executionDAO = executionDAO;
+        this.metricsCollector = metricsCollector;
+        this.systemTaskRegistry = systemTaskRegistry;
+        this.executionLockService = executionLockService;
+        this.properties = properties;
+        this.workflowExecutor = workflowExecutor;
+    }
+
+    @Async(SWEEPER_EXECUTOR_NAME)
+    public CompletableFuture<Void> sweepAsync(String workflowId) {
+        metricsCollector.getTimer("workflowSweeper").record(() -> sweep(workflowId));
+        return CompletableFuture.completedFuture(null);
+    }
+
+    private void sweep(String workflowId) {
+        boolean workflowLocked = false;
+        try {
+            workflowLocked = executionLockService.acquireLock(workflowId);
+            if (!workflowLocked) {
+                return;
+            }
+            log.info("Running sweeper for workflow {}, acquired lock", workflowId);
+            // 1. Run decide on the workflow
+            WorkflowModel workflow = executionDAO.getWorkflow(workflowId, true);
+            if (workflow == null) {
+                throw new NotFoundException(
```

Comment:
Throwing the exception here and catching directly in the catch block will make the execution slower. Can't we just remove the message here and return?

Reply:
Yes, will make the change to remove from `DECIDER_QUEUE` directly and return.